File: Programmers/야근지수.py
def solution(n, works):
    if n >= sum(works):
        return 0
    
    answer = 0
    
    import heapq
    h = []
    for work in works:
        heapq.heappush(h, -work)
    
    for i in range(n):
        value = heapq.heappop(h)
        value += 1
        heapq.heappush(h, value)
    
    for j in range(len(h)):
        h_value = heapq.heappop(h)
        answer += h_value * h_value
    
    return answer

File: JavaScript/야근지수.js
//효율성 통과x

function solution(n, works) {
    const r = (acc, curr) => acc + curr;
    const sum = works.reduce(r);
    if (sum <= n) return 0;
    let answer = 0;
    for (let i = 0; i < n; i++) works[works.indexOf(Math.max(...works))]--;
    for (let j = 0; j < works.length; j++) answer += works[j]**2;
    return answer;
}

console.log(solution(4, [4, 3, 3]));